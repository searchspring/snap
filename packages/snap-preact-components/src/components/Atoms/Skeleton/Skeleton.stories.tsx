import { h } from 'preact';

import { ArgsTable, PRIMARY_STORY, Markdown } from '@storybook/blocks';

import { componentArgs, highlightedCode } from '../../../utilities';
import { Skeleton, SkeletonProps } from './Skeleton';
import Readme from '../Skeleton/readme.md';

export default {
	title: 'Atoms/Skeleton',
	component: Skeleton,
	tags: ['autodocs'],
	parameters: {
		docs: {
			page: () => (
				<div>
					<Markdown
						options={{
							overrides: {
								code: highlightedCode,
							},
						}}
					>
						{Readme}
					</Markdown>
					<ArgsTable story={PRIMARY_STORY} />
				</div>
			),
		},
	},
	decorators: [
		(Story: any) => (
			<div
				style={{
					width: '200px',
					height: '200px',
					position: 'relative',
				}}
			>
				<Story height="200px" />
			</div>
		),
	],
	argTypes: {
		height: {
			description: 'CSS sizing to set the height of the Skeleton',
			defaultValue: '100px',
			type: { required: true },
			table: {
				type: {
					summary: 'string',
				},
			},
			control: { type: 'text' },
		},
		width: {
			description: 'CSS sizing to set the width of the Skeleton',
			defaultValue: '100px',
			type: { required: true },
			table: {
				type: {
					summary: 'string',
				},
			},
			control: { type: 'text' },
		},
		round: {
			table: {
				type: {
					summary: 'boolean',
				},
			},
			control: { type: 'boolean' },
		},
		backgroundColor: {
			description: 'Background color',
			defaultValue: '#ebebeb',
			table: {
				type: {
					summary: 'string',
				},
				defaultValue: { summary: 'string' },
			},
			control: { type: 'text' },
		},
		animatedColor: {
			description: 'Color setting for the vertical animated bar',
			defaultValue: '#f5f5f5',
			table: {
				type: {
					summary: 'string',
				},
				defaultValue: { summary: 'string' },
			},
			control: { type: 'text' },
		},
		...componentArgs,
	},
};

export const Default = (args: SkeletonProps) => <Skeleton {...args} />;
Default.args = {
	width: '100px',
	height: '100px',
};

export const Circle = (args: SkeletonProps) => <Skeleton {...args} />;
Circle.args = {
	width: '100px',
	height: '100px',
	round: true,
};
