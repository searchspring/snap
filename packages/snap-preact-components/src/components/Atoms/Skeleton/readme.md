# Skeleton

Renders a div with a animation for use when building loading skeletons. 

## Usage

### Height
The `height` prop specifies the css height of the skeleton div in px.

```jsx
<Skeleton height={200} width={150}/>
```

### Width

The `width` prop specifies the css width of the skeleton div in px.

```jsx
<Skeleton height={200} width={150}/>
```

### Round

The `round` prop is used for when you are wanting to render a circle instead of a block.

```jsx
<Skeleton height={200} width={150}/>
```

### Bgcolor

The `bgcolor` prop is used for changing the background color of the skeleton.

```jsx
<Skeleton height={200} width={150} bgcolor={'#333333'}/>
```