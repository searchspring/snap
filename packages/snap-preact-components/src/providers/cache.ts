export { CacheProvider } from '@emotion/react';
import createCache from '@emotion/cache';

export const cache = createCache({
	key: 'ss',
	prepend: true,
});
