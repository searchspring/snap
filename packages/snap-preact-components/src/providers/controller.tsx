import { jsx } from '@emotion/react';
import preact, { h, createContext, ComponentChildren, ComponentType } from 'preact';
import { useContext } from 'preact/hooks';
import type { AbstractController } from '@searchspring/snap-controller';

const ControllerContext = createContext<null | AbstractController>(null);

export const ControllerProvider = ({ children, controller }: { children: ComponentChildren; controller: AbstractController }) => {
	return <ControllerContext.Provider value={controller}>{children}</ControllerContext.Provider>;
};

export const useController = () => useContext(ControllerContext);

export const withController = (Component: ComponentType) => (props: any) => {
	return <Component controller={useController()} {...props} />;
};
