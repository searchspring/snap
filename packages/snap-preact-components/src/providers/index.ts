export * from './store';
export * from './theme';
