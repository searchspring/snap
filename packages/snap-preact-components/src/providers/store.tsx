import preact, { h, createContext, ComponentChildren, ComponentType } from 'preact';
import { useContext } from 'preact/hooks';
import type { AbstractStore } from '@searchspring/snap-store-mobx';

const StoreContext = createContext<null | AbstractStore>(null);

export const StoreProvider = ({ children, store }: { children: ComponentChildren; store: AbstractStore }) => {
	return <StoreContext.Provider value={store}>{children}</StoreContext.Provider>;
};

export const useStore = () => useContext(StoreContext);

export function withStore<C extends ComponentType>(Component: C): C {
	return ((props: any) => <StoreContext.Consumer>{(store) => <Component {...props} store={store} />}</StoreContext.Consumer>) as C;
}
