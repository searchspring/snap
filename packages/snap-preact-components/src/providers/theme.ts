import { ThemeProvider as EmotionTheme } from '@emotion/react';
export { css, useTheme, withTheme } from '@emotion/react';

interface ThemeProviderProps {
	theme: Partial<Theme> | ((outerTheme: Theme) => Theme);
	children?: any;
}
interface ThemeProvider {
	(props: ThemeProviderProps): React.ReactElement;
}
export const ThemeProvider: ThemeProvider = EmotionTheme;

export const defaultTheme: Theme = {
	colors: {
		primary: '#3A23AD',
		secondary: '#00cee1',
		hover: '#f8f6fd',
		text: {
			secondary: '#ffffff',
		},
		message: {
			error: '#cc1212',
			warning: '#ecaa15',
			info: '#4c3ce2',
		},
	},
	components: {},
};
export interface Theme {
	variables?: {
		[key: string]: unknown;
	};
	colors?: {
		primary: string;
		secondary: string;
		hover?: string;
		text?: {
			secondary: string;
		};
		message?: {
			error: string;
			warning: string;
			info: string;
		};
	};
	components?: any;
	namedComponents?: any;
}
