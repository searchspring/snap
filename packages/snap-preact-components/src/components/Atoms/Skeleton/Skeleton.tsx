/** @jsx jsx */
import { h } from 'preact';

import { jsx, css } from '@emotion/react';
import classnames from 'classnames';
import { observer } from 'mobx-react-lite';

import { Theme, useTheme, CacheProvider } from '../../../providers';
import { ComponentProps } from '../../../types';
import { CSSProperties } from 'react';

const CSS = {
	skeleton: ({ width, height, round, bgcolor }) =>
		css({
			'@keyframes react-loading-skeleton': {
				from: {
					transform: 'translateX(-100%)',
				},
				to: {
					transform: 'translateX(100%)',
				},
			},

			width: `${width}px`,
			height: `${height}px`,
			borderRadius: round ? `${width}px` : '0.25rem',

			backgroundColor: bgcolor,
			display: 'inline-flex',
			lineHeight: '1',

			position: 'relative',
			overflow: 'hidden',
			zIndex: '1' as CSSProperties['zIndex'] /* Necessary for overflow: hidden to work correctly in Safari */,

			'&:after': {
				content: '""',
				display: 'block',
				position: 'absolute',
				left: '0',
				right: '0',
				height: '100%',
				backgroundRepeat: 'no-repeat',
				backgroundImage: `linear-gradient(90deg, ${bgcolor}, #f5f5f5, ${bgcolor})`,
				transform: 'translateX(-100%)',
				animationName: 'react-loading-skeleton',
				animationDirection: 'normal',
				animationDuration: '1.5s',
				animationTimingFunction: 'ease-in-out',
				animationIterationCount: 'infinite',
			},
		}),
};

export const Skeleton = observer((properties: SkeletonProps): JSX.Element => {
	const globalTheme: Theme = useTheme();

	const props: SkeletonProps = {
		// default props
		bgcolor: '#ebebeb',
		// global theme
		...globalTheme?.components?.skeleton,
		// props
		...properties,
		...properties.theme?.components?.skeleton,
	};
	const { width, height, round, bgcolor, disableStyles, className, style } = props;

	const styling: { css?: any } = {};
	if (!disableStyles) {
		styling.css = [CSS.skeleton({ width, height, round, bgcolor }), style];
	} else if (style) {
		styling.css = [style];
	}
	return (
		<CacheProvider>
			<div {...styling} className={classnames('ss__skeleton', className)}></div>
		</CacheProvider>
	);
});

export interface SkeletonProps extends ComponentProps {
	width: number;
	height: number;
	round?: boolean;
	bgcolor?: string;
}
