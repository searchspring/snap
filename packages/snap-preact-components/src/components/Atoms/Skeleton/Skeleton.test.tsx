import { h } from 'preact';

import { render } from '@testing-library/preact';
import { ThemeProvider } from '../../../providers';

import { Skeleton } from './Skeleton';

describe('Skeleton Component', () => {
	it('renders', () => {
		const rendered = render(<Skeleton height="200px" width="200px" />);
		const skeleton = rendered.container.querySelector('.ss__skeleton');
		expect(skeleton).toBeInTheDocument();
	});

	it('has correct demmensions', () => {
		const rendered = render(<Skeleton height="400px" width="300px" />);
		const skeleton = rendered.container.querySelector('.ss__skeleton');
		expect(skeleton).toHaveStyle('height:400px; width:300px');
	});

	it('can change the backgroundColor color', () => {
		const rendered = render(<Skeleton height="400px" width="300px" backgroundColor={'red'} />);
		const skeleton = rendered.container.querySelector('.ss__skeleton');
		expect(skeleton).toHaveStyle('background-color:red');
	});

	it('can change to a circle', () => {
		const rendered = render(<Skeleton height="300px" width="300px" round={true} />);
		const skeleton = rendered.container.querySelector('.ss__skeleton');
		expect(skeleton).toHaveStyle('border-radius:300px');
	});

	it('renders with classname', () => {
		const className = 'classy';
		const rendered = render(<Skeleton className={className} height="300px" width="300px" />);
		const skeleton = rendered.container.querySelector('.ss__skeleton');

		expect(skeleton).toHaveClass(className);
	});

	it('disables styles', () => {
		const rendered = render(<Skeleton disableStyles height="300px" width="300px" />);
		const skeleton = rendered.container.querySelector('.ss__skeleton');

		expect(skeleton?.classList).toHaveLength(1);
	});

	expect(true).toBe(true);
});

describe('Skeleton theming works', () => {
	it('is themeable with ThemeProvider', () => {
		const globalTheme = {
			components: {
				skeleton: {
					disableStyles: true,
				},
			},
		};
		const rendered = render(
			<ThemeProvider theme={globalTheme}>
				<Skeleton height="300px" width="300px" />
			</ThemeProvider>
		);
		const skeleton = rendered.container.querySelector('.ss__skeleton');
		expect(skeleton).toBeInTheDocument();
		expect(skeleton?.classList.length).toBe(1);
	});

	it('is themeable with theme prop', () => {
		const propTheme = {
			components: {
				skeleton: {
					disableStyles: true,
				},
			},
		};
		const rendered = render(<Skeleton height="300px" width="300px" theme={propTheme} />);
		const skeleton = rendered.container.querySelector('.ss__skeleton');
		expect(skeleton).toBeInTheDocument();
		expect(skeleton?.classList.length).toBe(1);
	});

	it('is theme prop overrides ThemeProvider', () => {
		const globalTheme = {
			components: {
				skeleton: {
					disableStyles: false,
				},
			},
		};
		const propTheme = {
			components: {
				skeleton: {
					disableStyles: true,
				},
			},
		};
		const rendered = render(
			<ThemeProvider theme={globalTheme}>
				<Skeleton height="300px" width="300px" theme={propTheme} />
			</ThemeProvider>
		);

		const skeleton = rendered.container.querySelector('.ss__skeleton');
		expect(skeleton).toBeInTheDocument();
		expect(skeleton?.classList.length).toBe(1);
	});
});
