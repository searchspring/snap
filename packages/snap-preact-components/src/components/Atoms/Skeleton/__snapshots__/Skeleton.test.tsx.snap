// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton Component can change the backgroundColor color 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  from {
    -webkit-transform: translateX(-100%);
    -moz-transform: translateX(-100%);
    -ms-transform: translateX(-100%);
    transform: translateX(-100%);
  }

  to {
    -webkit-transform: translateX(100%);
    -moz-transform: translateX(100%);
    -ms-transform: translateX(100%);
    transform: translateX(100%);
  }
}

@keyframes animation-0 {
  from {
    -webkit-transform: translateX(-100%);
    -moz-transform: translateX(-100%);
    -ms-transform: translateX(-100%);
    transform: translateX(-100%);
  }

  to {
    -webkit-transform: translateX(100%);
    -moz-transform: translateX(100%);
    -ms-transform: translateX(100%);
    transform: translateX(100%);
  }
}

@keyframes animation-0 {
  from {
    -webkit-transform: translateX(-100%);
    -moz-transform: translateX(-100%);
    -ms-transform: translateX(-100%);
    transform: translateX(-100%);
  }

  to {
    -webkit-transform: translateX(100%);
    -moz-transform: translateX(100%);
    -ms-transform: translateX(100%);
    transform: translateX(100%);
  }
}

@keyframes animation-0 {
  from {
    -webkit-transform: translateX(-100%);
    -moz-transform: translateX(-100%);
    -ms-transform: translateX(-100%);
    transform: translateX(-100%);
  }

  to {
    -webkit-transform: translateX(100%);
    -moz-transform: translateX(100%);
    -ms-transform: translateX(100%);
    transform: translateX(100%);
  }
}

.emotion-0 {
  width: 300px;
  height: 400px;
  border-radius: 0.25rem;
  background-color: red;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  line-height: 1;
  position: relative;
  overflow: hidden;
  z-index: 1;
}

.emotion-0:after {
  content: "";
  display: block;
  position: absolute;
  left: 0;
  right: 0;
  height: 100%;
  background-repeat: no-repeat;
  background-image: linear-gradient(90deg, red, #f5f5f5, red);
  -webkit-transform: translateX(-100%);
  -moz-transform: translateX(-100%);
  -ms-transform: translateX(-100%);
  transform: translateX(-100%);
  -webkit-animation: animation-0 1.5s linear infinite;
  animation: animation-0 1.5s linear infinite;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
}

<div
    class="ss__skeleton emotion-0"
  />
</DocumentFragment>
`;

exports[`Skeleton Component can change to a circle 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  from {
    -webkit-transform: translateX(-100%);
    -moz-transform: translateX(-100%);
    -ms-transform: translateX(-100%);
    transform: translateX(-100%);
  }

  to {
    -webkit-transform: translateX(100%);
    -moz-transform: translateX(100%);
    -ms-transform: translateX(100%);
    transform: translateX(100%);
  }
}

@keyframes animation-0 {
  from {
    -webkit-transform: translateX(-100%);
    -moz-transform: translateX(-100%);
    -ms-transform: translateX(-100%);
    transform: translateX(-100%);
  }

  to {
    -webkit-transform: translateX(100%);
    -moz-transform: translateX(100%);
    -ms-transform: translateX(100%);
    transform: translateX(100%);
  }
}

@keyframes animation-0 {
  from {
    -webkit-transform: translateX(-100%);
    -moz-transform: translateX(-100%);
    -ms-transform: translateX(-100%);
    transform: translateX(-100%);
  }

  to {
    -webkit-transform: translateX(100%);
    -moz-transform: translateX(100%);
    -ms-transform: translateX(100%);
    transform: translateX(100%);
  }
}

@keyframes animation-0 {
  from {
    -webkit-transform: translateX(-100%);
    -moz-transform: translateX(-100%);
    -ms-transform: translateX(-100%);
    transform: translateX(-100%);
  }

  to {
    -webkit-transform: translateX(100%);
    -moz-transform: translateX(100%);
    -ms-transform: translateX(100%);
    transform: translateX(100%);
  }
}

@keyframes animation-0 {
  from {
    -webkit-transform: translateX(-100%);
    -moz-transform: translateX(-100%);
    -ms-transform: translateX(-100%);
    transform: translateX(-100%);
  }

  to {
    -webkit-transform: translateX(100%);
    -moz-transform: translateX(100%);
    -ms-transform: translateX(100%);
    transform: translateX(100%);
  }
}

.emotion-0 {
  width: 300px;
  height: 300px;
  border-radius: 300px;
  background-color: #ebebeb;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  line-height: 1;
  position: relative;
  overflow: hidden;
  z-index: 1;
}

.emotion-0:after {
  content: "";
  display: block;
  position: absolute;
  left: 0;
  right: 0;
  height: 100%;
  background-repeat: no-repeat;
  background-image: linear-gradient(90deg, #ebebeb, #f5f5f5, #ebebeb);
  -webkit-transform: translateX(-100%);
  -moz-transform: translateX(-100%);
  -ms-transform: translateX(-100%);
  transform: translateX(-100%);
  -webkit-animation: animation-0 1.5s linear infinite;
  animation: animation-0 1.5s linear infinite;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
}

<div
    class="ss__skeleton emotion-0"
  />
</DocumentFragment>
`;

exports[`Skeleton Component disables styles 1`] = `
<DocumentFragment>
  <div
    class="ss__skeleton"
  />
</DocumentFragment>
`;

exports[`Skeleton Component has correct demmensions 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  from {
    -webkit-transform: translateX(-100%);
    -moz-transform: translateX(-100%);
    -ms-transform: translateX(-100%);
    transform: translateX(-100%);
  }

  to {
    -webkit-transform: translateX(100%);
    -moz-transform: translateX(100%);
    -ms-transform: translateX(100%);
    transform: translateX(100%);
  }
}

@keyframes animation-0 {
  from {
    -webkit-transform: translateX(-100%);
    -moz-transform: translateX(-100%);
    -ms-transform: translateX(-100%);
    transform: translateX(-100%);
  }

  to {
    -webkit-transform: translateX(100%);
    -moz-transform: translateX(100%);
    -ms-transform: translateX(100%);
    transform: translateX(100%);
  }
}

@keyframes animation-0 {
  from {
    -webkit-transform: translateX(-100%);
    -moz-transform: translateX(-100%);
    -ms-transform: translateX(-100%);
    transform: translateX(-100%);
  }

  to {
    -webkit-transform: translateX(100%);
    -moz-transform: translateX(100%);
    -ms-transform: translateX(100%);
    transform: translateX(100%);
  }
}

.emotion-0 {
  width: 300px;
  height: 400px;
  border-radius: 0.25rem;
  background-color: #ebebeb;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  line-height: 1;
  position: relative;
  overflow: hidden;
  z-index: 1;
}

.emotion-0:after {
  content: "";
  display: block;
  position: absolute;
  left: 0;
  right: 0;
  height: 100%;
  background-repeat: no-repeat;
  background-image: linear-gradient(90deg, #ebebeb, #f5f5f5, #ebebeb);
  -webkit-transform: translateX(-100%);
  -moz-transform: translateX(-100%);
  -ms-transform: translateX(-100%);
  transform: translateX(-100%);
  -webkit-animation: animation-0 1.5s linear infinite;
  animation: animation-0 1.5s linear infinite;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
}

<div
    class="ss__skeleton emotion-0"
  />
</DocumentFragment>
`;

exports[`Skeleton Component renders 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  from {
    -webkit-transform: translateX(-100%);
    -moz-transform: translateX(-100%);
    -ms-transform: translateX(-100%);
    transform: translateX(-100%);
  }

  to {
    -webkit-transform: translateX(100%);
    -moz-transform: translateX(100%);
    -ms-transform: translateX(100%);
    transform: translateX(100%);
  }
}

@keyframes animation-0 {
  from {
    -webkit-transform: translateX(-100%);
    -moz-transform: translateX(-100%);
    -ms-transform: translateX(-100%);
    transform: translateX(-100%);
  }

  to {
    -webkit-transform: translateX(100%);
    -moz-transform: translateX(100%);
    -ms-transform: translateX(100%);
    transform: translateX(100%);
  }
}

.emotion-0 {
  width: 200px;
  height: 200px;
  border-radius: 0.25rem;
  background-color: #ebebeb;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  line-height: 1;
  position: relative;
  overflow: hidden;
  z-index: 1;
}

.emotion-0:after {
  content: "";
  display: block;
  position: absolute;
  left: 0;
  right: 0;
  height: 100%;
  background-repeat: no-repeat;
  background-image: linear-gradient(90deg, #ebebeb, #f5f5f5, #ebebeb);
  -webkit-transform: translateX(-100%);
  -moz-transform: translateX(-100%);
  -ms-transform: translateX(-100%);
  transform: translateX(-100%);
  -webkit-animation: animation-0 1.5s linear infinite;
  animation: animation-0 1.5s linear infinite;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
}

<div
    class="ss__skeleton emotion-0"
  />
</DocumentFragment>
`;

exports[`Skeleton Component renders with classname 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  from {
    -webkit-transform: translateX(-100%);
    -moz-transform: translateX(-100%);
    -ms-transform: translateX(-100%);
    transform: translateX(-100%);
  }

  to {
    -webkit-transform: translateX(100%);
    -moz-transform: translateX(100%);
    -ms-transform: translateX(100%);
    transform: translateX(100%);
  }
}

@keyframes animation-0 {
  from {
    -webkit-transform: translateX(-100%);
    -moz-transform: translateX(-100%);
    -ms-transform: translateX(-100%);
    transform: translateX(-100%);
  }

  to {
    -webkit-transform: translateX(100%);
    -moz-transform: translateX(100%);
    -ms-transform: translateX(100%);
    transform: translateX(100%);
  }
}

@keyframes animation-0 {
  from {
    -webkit-transform: translateX(-100%);
    -moz-transform: translateX(-100%);
    -ms-transform: translateX(-100%);
    transform: translateX(-100%);
  }

  to {
    -webkit-transform: translateX(100%);
    -moz-transform: translateX(100%);
    -ms-transform: translateX(100%);
    transform: translateX(100%);
  }
}

@keyframes animation-0 {
  from {
    -webkit-transform: translateX(-100%);
    -moz-transform: translateX(-100%);
    -ms-transform: translateX(-100%);
    transform: translateX(-100%);
  }

  to {
    -webkit-transform: translateX(100%);
    -moz-transform: translateX(100%);
    -ms-transform: translateX(100%);
    transform: translateX(100%);
  }
}

@keyframes animation-0 {
  from {
    -webkit-transform: translateX(-100%);
    -moz-transform: translateX(-100%);
    -ms-transform: translateX(-100%);
    transform: translateX(-100%);
  }

  to {
    -webkit-transform: translateX(100%);
    -moz-transform: translateX(100%);
    -ms-transform: translateX(100%);
    transform: translateX(100%);
  }
}

@keyframes animation-0 {
  from {
    -webkit-transform: translateX(-100%);
    -moz-transform: translateX(-100%);
    -ms-transform: translateX(-100%);
    transform: translateX(-100%);
  }

  to {
    -webkit-transform: translateX(100%);
    -moz-transform: translateX(100%);
    -ms-transform: translateX(100%);
    transform: translateX(100%);
  }
}

.emotion-0 {
  width: 300px;
  height: 300px;
  border-radius: 0.25rem;
  background-color: #ebebeb;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  line-height: 1;
  position: relative;
  overflow: hidden;
  z-index: 1;
}

.emotion-0:after {
  content: "";
  display: block;
  position: absolute;
  left: 0;
  right: 0;
  height: 100%;
  background-repeat: no-repeat;
  background-image: linear-gradient(90deg, #ebebeb, #f5f5f5, #ebebeb);
  -webkit-transform: translateX(-100%);
  -moz-transform: translateX(-100%);
  -ms-transform: translateX(-100%);
  transform: translateX(-100%);
  -webkit-animation: animation-0 1.5s linear infinite;
  animation: animation-0 1.5s linear infinite;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
}

<div
    class="ss__skeleton classy emotion-0"
  />
</DocumentFragment>
`;

exports[`Skeleton Component uses bocachica theme 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  from {
    -webkit-transform: translateX(-100%);
    -moz-transform: translateX(-100%);
    -ms-transform: translateX(-100%);
    transform: translateX(-100%);
  }

  to {
    -webkit-transform: translateX(100%);
    -moz-transform: translateX(100%);
    -ms-transform: translateX(100%);
    transform: translateX(100%);
  }
}

.emotion-0 {
  width: 200px;
  height: 200px;
  border-radius: 200px;
  background-color: red;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  line-height: 1;
  position: relative;
  overflow: hidden;
  z-index: 1;
}

.emotion-0:after {
  content: "";
  display: block;
  position: absolute;
  left: 0;
  right: 0;
  height: 100%;
  background-repeat: no-repeat;
  background-image: linear-gradient(90deg, red, #f5f5f5, red);
  -webkit-transform: translateX(-100%);
  -moz-transform: translateX(-100%);
  -ms-transform: translateX(-100%);
  transform: translateX(-100%);
  -webkit-animation: animation-0 1.5s linear infinite;
  animation: animation-0 1.5s linear infinite;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
}

<div
    class="ss__skeleton emotion-0"
  />
</DocumentFragment>
`;

exports[`Skeleton theming works is theme prop overrides ThemeProvider 1`] = `
<DocumentFragment>
  <div
    class="ss__skeleton"
  />
</DocumentFragment>
`;

exports[`Skeleton theming works is themeable with ThemeProvider 1`] = `
<DocumentFragment>
  <div
    class="ss__skeleton"
  />
</DocumentFragment>
`;

exports[`Skeleton theming works is themeable with theme prop 1`] = `
<DocumentFragment>
  <div
    class="ss__skeleton"
  />
</DocumentFragment>
`;
